### ISSUE TITLE
Concern regarding possible resource leak

### ISSUE THREAD

**Issue by akovac35** [dotnet/EntityFramework.Docs#2886]
I have a concern regarding the following example:

```cs
var options = new DbContextOptionsBuilder<BloggingContext>()
    .UseSqlServer(new SqlConnection(connectionString))
    .Options;
```

We can see that a new connection object is passed the the DbContextOptionsBuilder - should this connection be disposed at some point? I have examined the ef core code but I couldn't find a confirmation that this provided connection is actually disposed at some point. So the example should be changed to mention that the connection should be disposed of at some point. If a factory is used, then this factory should track the connections in a weak reference and dispose them when itself is disposed.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1073c238-9389-d826-a696-f1108a4f2ce0
* Version Independent ID: cf56babd-def0-d76e-8eee-e33341b6a009
* Content: [Transactions - EF Core](https://docs.microsoft.com/en-us/ef/core/saving/transactions)
* Content Source: [entity-framework/core/saving/transactions.md](https://github.com/dotnet/EntityFramework.Docs/blob/master/entity-framework/core/saving/transactions.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @roji
* Microsoft Alias: **avickers**

**Comment by ajcvickers** [comment]
@akovac35 Yes, this is a leak. If the application creates the DbConnection, then the application is responsible for disposing it.

/cc @roji

**Comment by roji** [comment]
Was this me? :fearful: 

Pushed a fix into #2882
